Fixed decimal initial version

tnadrai opened the pull request:
graphs not working properly

tnadrai committed:
Fixed decimal initial version

szerintedmi commented:
Deploy preview for *augmintsim* ready!

Built with commit 410b6dd024dcf1d72b8a1ec6a85dbb2061f25980

https://deploy-preview-92--augmintsim.netlify.com

tnadrai committed:
Merge branch 'staging' into fixed_decimal

szerintedmi commented:
@CryppTom : my initial guess is that your class' getter is not working or used incorrectly at some places. If you look at the log there are entries where acd amount is shown like this:
`"acdAmount":{"number":586.632}`

tnadrai committed:
graphs start to appear, until day 60 then "AugmintError: exchangeAcd has gone negative" exception shows up

tnadrai committed:
fixed decimal - first working version - graphs shown after day 60

tnadrai committed:
Merge branch 'staging' into fixed_decimal

szerintedmi commented on src/lib/fixedDecimal.js:
```diff
@@ -0,0 +1,106 @@
+    P.round = function(dp,rm) {
+        let Big = this.constructor;
+        let returnValue = new Big(this);
+        returnValue.number = Number(parseFloat(this.number).toFixed(dp));
```
why the parseFloat?  this.number should always be a number

szerintedmi commented on src/lib/fixedDecimal.js:
```diff
@@ -0,0 +1,106 @@
+    P.add = function(number) {
+        let Big = this.constructor;
+        let returnValue = new Big(this);
+        returnValue.number =  Number(parseFloat(this.number)+number).toFixed(Big.DP);
```
closing parentheses seems to be at the wrong place here, `.toFixed` returns string:
```
Number(parseFloat(this.number)+number).toFixed(Big.DP)
```
maybe that's why you needed to hack with parseFloats ?

szerintedmi commented on src/lib/bigNums.js:
```diff
@@ -1,5 +1,5 @@
 "use strict";
-const Big = require("big.js");
```
we can now remove big.js from package.json and regenerate yarn.lock

szerintedmi commented on src/lib/bigNums.js:
```diff
@@ -1,5 +1,5 @@
 "use strict";
-const Big = require("big.js");
+const Big = require("../lib/fixedDecimal.js");
```
housekeeping for future readability:  "Big" variable should be renamed to FixedDecimal (maybe the class name should be FixedDecimal too, capital indicating it's a class)

szerintedmi commented on src/lib/bigNums.js:
```diff
@@ -1,5 +1,5 @@
 "use strict";
-const Big = require("big.js");
+const Big = require("../lib/fixedDecimal.js");
 const BigAcd = Big(); // ACD values
```
nitpicking for future readability : could we easily do a search & replace for BigAcd -->  FixedAcd , BigEth -> FixedEth , BigPt -> FixedPt in the whole codebase?

szerintedmi reviewed (changes requested):
looks great, left a few final polishing change suggestions

tnadrai committed:
fixed decimal enhancements:
-removed parseFloat
-changed parenthesis
-Big renamed to FixedDecimal
-renamed varibles BigAcd --> FixedAcd , BigEth -> FixedEth , BigPt -> FixedPt

tnadrai committed:
Merge branch 'fixed_decimal' of https://github.com/Augmint/augmint-sim into fixed_decimal

tnadrai committed:
removing big from package.json and regeneration of yarn.lock
